### PR TITLE
Add release scripts for modules and binaries

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

File: releasing/releaseall.sh [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    go mod tidy
+    go test ./...
+    go mod tidy
+    git add .
+    git commit -m "update go.mod for release" || echo "no changes to go.mod"
+  fi
+
+  echo "Push $module branch: $branch and tag $tag?"
+  read answer
+  if [ "$answer" != "${answer#[Yy]}" ] ;then
```
[the remark this answers is not in the document]
Comment (pwittrock) [reply releasing/releaseall.sh]: how about env vars?

File: releasing/releaseall.sh [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  # create a workspace for the new release to be cut
+  tw=$(mktemp -d /tmp/kustomize-releases-XXXXXX)
+  echo "release workspace: $tw"
+  git branch -f $branch upstream/master # always release from master
+  git worktree add $tw $branch
+  pushd .
+  cd $tw
+
+  # fix the go mod replaces if they exist
+  cd $module
```
[the remark this answers is not in the document]
Comment (pwittrock) [reply releasing/releaseall.sh]: done

File: releasing/releaseall.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (monopole) [reply releasing/releaseall.sh]: whatever

**Comment by pwittrock** [comment]
Will follow up with a doc update

**Pull request merged**